rust-symcrypt/src/hmac.rs

```diff
@@ -0,0 +1,215 @@
+use core::ffi::c_void;
+use std::mem;
+use std::ptr;
+use symcrypt_sys;
+
+pub const SHA256_HMAC_RESULT_SIZE: usize = symcrypt_sys::SYMCRYPT_SHA256_RESULT_SIZE as usize;
+pub const SHA384_HMAC_RESULT_SIZE: usize = symcrypt_sys::SYMCRYPT_SHA384_RESULT_SIZE as usize;
+
+pub struct HmacSha256State {
```

Comment:
Same comment here regarding whether macros could/should be used here to reduce repetition in these implementations.